examples_tests/09.ColorSpaceTest/ApplicationHandler.cpp

```diff
@@ -0,0 +1,230 @@
+{
+	auto samplerDescriptorSet3 = driver->createGPUDescriptorSet(core::smart_refctd_ptr(gpuDescriptorSetLayout3));
+
+	IGPUDescriptorSet::SDescriptorInfo info;
+	{
+		info.desc = std::move(gpuImageView);
+		ISampler::SParams samplerParams = { ISampler::ETC_CLAMP_TO_EDGE, ISampler::ETC_CLAMP_TO_EDGE, ISampler::ETC_CLAMP_TO_EDGE, ISampler::ETBC_FLOAT_OPAQUE_BLACK, ISampler::ETF_LINEAR, ISampler::ETF_LINEAR, ISampler::ESMM_LINEAR, 0u, false, ECO_ALWAYS };
+		info.image.sampler = driver->createGPUSampler(samplerParams);
+		info.image.imageLayout = EIL_SHADER_READ_ONLY_OPTIMAL;
+	}
+
+	IGPUDescriptorSet::SWriteDescriptorSet write;
+	write.dstSet = samplerDescriptorSet3.get();
+	write.binding = 0u;
+	write.arrayElement = 0u;
+	write.count = 1u;
+	write.descriptorType = EDT_COMBINED_IMAGE_SAMPLER;
+	write.info = &info;
+
+	driver->updateDescriptorSets(1u, &write, 0u, nullptr);
+
+	std::wostringstream characterStream;
+	characterStream << L"Color Space Test Demo - Irrlicht Engine [" << driver->getName() << "] - CURRENT IMAGE: " << currentHandledImageFileName.c_str() << " - EXTENSION: " << currentHandledImageExtension.c_str();
+	device->setWindowCaption(characterStream.str().c_str());
+
+	auto startPoint = std::chrono::high_resolution_clock::now();
+
+	while (device->run())
+	{
+		auto aPoint = std::chrono::high_resolution_clock::now();
+		if (std::chrono::duration_cast<std::chrono::milliseconds>(aPoint - startPoint).count() > SWITCH_IMAGES_PER_X_MILISECONDS)
+			break;
+
+		driver->beginScene(true, true);
+
+		driver->bindGraphicsPipeline(currentGpuPipeline.get());
+		driver->bindDescriptorSets(EPBP_GRAPHICS, currentGpuPipeline->getLayout(), 3u, 1u, &samplerDescriptorSet3.get(), nullptr);
+		driver->drawMeshBuffer(currentGpuMeshBuffer.get());
+
+		driver->blitRenderTargets(nullptr, screenShotFrameBuffer, false, false);
+		driver->endScene();
+	}
+
+	ext::ScreenShot::createScreenShoot(device, screenShotFrameBuffer->getAttachment(video::EFAP_COLOR_ATTACHMENT0), "screenShot_" + currentHandledImageFileName + ".png");
+}
+
+void ApplicationHandler::performImageTest(std::string path)
+{
+	os::Printer::log("Reading", path);
+
+	auto assetManager = device->getAssetManager();
+
+	smart_refctd_ptr<ICPUImageView> cpuImageView;
+
+	auto cpuTexture = assetManager->getAsset(path, {});
+	auto cpuTextureContents = cpuTexture.getContents();
+	
+	if (cpuTextureContents.first == cpuTextureContents.second)
+	{
+		os::Printer::log("CANNOT PERFORM THE TEST - SKIPPING. LOADING WENT WRONG", ELL_WARNING);
+		return;
+	}
+
+	io::path filename, extension, finalFileNameWithExtension;
+	core::splitFilename(path.c_str(), nullptr, &filename, &extension);
+	finalFileNameWithExtension = filename + ".";
+	finalFileNameWithExtension += extension;
+
+	// Can't do that because write is still broken
+	bool writeable = (extension != "dds");
```

Comment:
its actually writeable cause GLI supports writing it

Reply:
I told u that I got some weird crashes, anyway now we need to the bug firstly